install/0000_30_machine-api-operator_09_rbac.yaml

```diff
@@ -269,3 +269,21 @@ subjects:
   - kind: ServiceAccount
     name: machine-api-operator
     namespace: openshift-machine-api
+
+---
```

Comment:
IIUC it should happen later in the deployment because the `openshift-monitoring` namespace won't exist yet (the cluster monitoring operator runs at the 50 run-level) with the other [OLM](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md#how-do-i-get-added-as-a-special-run-level) stuff. See the [cluster-kube-apiserver-operator](https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/manifests/0000_90_kube-apiserver-operator_01_prometheusrole.yaml) for instance.

You also need to [bind](https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/manifests/0000_90_kube-apiserver-operator_02_prometheusrolebinding.yaml) this role to the `prometheus-k8s` service account.

Reply:
Thanks Simon, going to look

Reply:
I've followed the example from the monitoring guide [1], would it be ok to move this rbac to a specific file so I can change the level according to OLM etc? Also, if I move to other level I would have to move the `Service`, `ServiceMonitor` and `PrometheusRule` right?


[1] https://github.com/openshift/cluster-monitoring-operator/blob/23201e012586d4864ca23593621f843179c47412/assets/prometheus-k8s/role-specific-namespaces.yaml#L35-L50

Reply:
@iurygregory afaik it's fine to separate the rbac definition in a separate file as long as it stays in the same runlevel. your statement regarding the service, servicemonitor, and prometheus rules are correct :+1:

Reply:
@s-urbaniak Thanks for your review, the rbac would need to change to `kind: RoleBinding` ?

Reply:
@iurygregory you actually need both:

1. a role that refers to the services/endpoints/pods resources in your namespace and the allowed verbs get/list/watch (random example https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/manifests/0000_90_kube-apiserver-operator_01_prometheusrole.yaml)
2. a role binding which associates the above role with the `prometheus-k8s` service account (random example https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/manifests/0000_90_kube-apiserver-operator_02_prometheusrolebinding.yaml)

Reply:
@s-urbaniak the `RoleBinding` level to run would be 90? or can be 30 also?

Reply:
@iurygregory as discussed out of band, I suggest runlevel 90, there is no monitoring stack yet in runlevel 30.